Apple support for RenderDocCmd DisplayRendererPreview

Zorro666 opened the pull request:
## Description

Apple implementation to create Cocoa window to use for displaying the replay image.

Reformatted the *.mm files in the project with clang-format (locally I did not have the clang-format git hook enabled for *.mm files).

Updated the script used by the CI to check code formatting in the project "util/clang_format_all.sh" to include *.m & *.mm files.

Updated the script used by the CI to work on Mac OS using basic find operation and doing the regular expression matching using grep -E.

Example of running the changes to the code formatting script without fixes to the existing files

https://github.com/Zorro666/renderdoc/runs/2281689074?check_suite_focus=true

Showing failures for

- renderdoc/driver/vulkan/vk_apple.mm b/renderdoc/driver/vulkan/vk_apple.mm
- renderdoc/os/posix/apple/apple_helpers.mm b/renderdoc/os/posix/apple/apple_helpers.mm 

## Testing

For the clang_format_all.sh change: tested on Linux compared the output that would be passed to xargs before and after the change to the find command usage and the results were identical.

Examples from local testing

Replaying a Vulkan capture

<img width="1392" alt="image" src="https://user-images.githubusercontent.com/39392/113338672-525d9280-9321-11eb-9d67-b80100912b2a.png">

----
Replaying a GL capture

<img width="1392" alt="image" src="https://user-images.githubusercontent.com/39392/113338894-a6687700-9321-11eb-8e49-251e61807f2c.png">

----

Zorro666 committed:
Apple implementation of DisplayRendererPreview

Zorro666 committed:
Reformatting *.mm files with clang-format

baldurk reviewed (changes requested):
This looks good, but can you fix the CI formatting check to ensure that a formatting commit like this isn't needed in the future?

Zorro666 committed:
Added *.m & *.mm file extensions

Zorro666 committed:
Modified find to work on Mac

Do basic find without regular expressions
Use grep -E to do the regular expression matching
Use awk to reconstruct an equivalent find -print0 output
